bindings/java/src/main/native/jucx_common_def.cc

```diff
@@ -181,12 +182,14 @@ void send_callback(void *request, ucs_status_t status)
     ucp_request_free(request);
 }
 
-jobject process_request(void *request, jobject callback)
+jobject process_request(void *request, jobject callback, jobject data_buf)
 {
     JNIEnv *env = get_jni_env();
     jclass jucx_request_cls = env->FindClass("org/ucx/jucx/UcxRequest");
     jmethodID constructor = env->GetMethodID(jucx_request_cls, "<init>", "()V");
     jobject jucx_request = env->NewObject(jucx_request_cls, constructor);
+    jfieldID field = env->GetFieldID(jucx_request_cls, "data", "Ljava/nio/ByteBuffer;");
+    env->SetObjectField(jucx_request, field, data_buf);
```

Comment:
use write barrier before signaling completion on the request ucs_memory_cpu_store_fence()